heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -566,10 +575,22 @@ public FullQuery end(int resolved, int failed, int cancelled) throws Exception {
 
             return Optional.empty();
         }
+
+        public Histogram getRowDensityHistogram() {
+            /* The data is gathered in an efficient ConcurrentHashMultiset, to allow for multiple
+             * threads writing with minimum blocking. Reading the data only happens at the end of
+             * the watched operation, so here we build the histogram.
+             */
+            final Histogram.Builder builder = Histogram.builder();
+            for (final Long value : rowDensityData.elementSet()) {
+                builder.add(value);
+            }
+            return builder.build();
+        }
     }
 
     @RequiredArgsConstructor
-    private static class QuotaWatcher implements FetchQuotaWatcher, RetainQuotaWatcher {
+    private static class ReadWatcher implements FetchQuotaWatcher, RetainQuotaWatcher {
```

Comment:
You're right :) The benefit of being able to move other things to ResultCollector. I've changed it back to QuotaWatcher now.